File: Simulator/Views/Card.vb
Public Class Card

    Sub New()

        ' Esta llamada es exigida por el diseñador.
        InitializeComponent()

        ' Agregue cualquier inicialización después de la llamada a InitializeComponent().

    End Sub

    Sub New(card As Models.Card)
        Me.New()

        Me.Bank = card.Bank
        Me.Number = card.Number
        Me.Exp = card.Exp

    End Sub

#Region "Properties"

    Private _Bank As String = My.Application.Info.ProductName
    Public Property Bank() As String
        Get
            Return _Bank
        End Get
        Set(ByVal value As String)
            _Bank = value
        End Set
    End Property

    Private _Number As String = "0000 0000 0000 0000"
    Public Property Number() As String
        Get
            Return _Number
        End Get
        Set(ByVal value As String)
            _Number = value
        End Set
    End Property

    Private _Exp As Date = Now.Date
    Public Property Exp() As Date
        Get
            Return _Exp
        End Get
        Set(ByVal value As Date)
            _Exp = value
        End Set
    End Property

    Private _Cirrus As Boolean = True
    Public Property Cirrus() As Boolean
        Get
            Return _Cirrus
        End Get
        Set(ByVal value As Boolean)
            _Cirrus = value
        End Set
    End Property

    Private _Maestro As Boolean = True
    Public Property Maestro() As Boolean
        Get
            Return _Maestro
        End Get
        Set(ByVal value As Boolean)
            _Maestro = value
        End Set
    End Property

#End Region

    Protected Overrides Sub OnPaint(e As PaintEventArgs)
        MyBase.OnPaint(e)
        Me.LabelBank.Text = Me.Bank
        Me.LabelNumber.Text = Me.Number
        Me.LabelExp.Text = Me.Exp.ToString("MM/yy")
        Me.PictureBoxCirrus.Visible = Me.Cirrus
        Me.PictureBoxMaestro.Visible = Me.Maestro
    End Sub

    Private Sub Card_MouseDown(sender As Object, e As MouseEventArgs) Handles Me.MouseDown, PictureBoxMaestro.MouseDown, PictureBoxCirrus.MouseDown, PictureBoxChip.MouseDown, LabelNumber.MouseDown, LabelExp.MouseDown, LabelBank.MouseDown
        Me.DoDragDrop(Me.Number, DragDropEffects.Copy)
    End Sub

End Class

File: Simulator-CSharp/Views/Card.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Drawing;
using System.Data;
using System.Linq;
using System.Text;
using System.Windows.Forms;

namespace ATM.Views
{
    public partial class Card : UserControl
    {
        public Card()
        {
            InitializeComponent();
        }

        public Card(Models.Card card) : this()
        {
            Bank = card.Bank;
            Number = card.Number;
            Exp = card.Exp;
        }

        #region Properties

        private string _Bank = System.Reflection.Assembly.GetExecutingAssembly().GetName().Name;
        public string Bank
        {
            get { return _Bank; }
            set { _Bank = value; }
        }

        private string _Number = "0000 0000 0000 0000";
        public string Number
        {
            get { return _Number; }
            set { _Number = value; }
        }

        private System.DateTime _Exp = DateTime.Now.Date;
        public System.DateTime Exp
        {
            get { return _Exp; }
            set { _Exp = value; }
        }

        private bool _Cirrus = true;
        public bool Cirrus
        {
            get { return _Cirrus; }
            set { _Cirrus = value; }
        }

        private bool _Maestro = true;
        public bool Maestro
        {
            get { return _Maestro; }
            set { _Maestro = value; }
        }

        #endregion

        protected override void OnPaint(PaintEventArgs e)
        {
            base.OnPaint(e);
            labelBank.Text = this.Bank;
            labelNumber.Text = this.Number;
            labelExp.Text = this.Exp.ToString("MM/yy");
            pictureBoxCirrus.Visible = this.Cirrus;
            pictureBoxMaestro.Visible = this.Maestro;
        }

        private void Card_MouseDown(object sender, MouseEventArgs e)
        {
            DoDragDrop(Number, DragDropEffects.Copy);
        }
    }
}
